web-client/src/app/utils/errors/global-error-handler.ts

```diff
@@ -0,0 +1,26 @@
+import { ErrorHandler, Injectable, NgZone } from '@angular/core';
+import { NavController } from '@ionic/angular';
+import { SwalHelper } from '../notification/swal-helper';
+
+@Injectable()
+export class GlobalErrorHandler implements ErrorHandler {
+  constructor(
+    private zone: NgZone,
+    private notification: SwalHelper,
+    private navCtrl: NavController
+  ) {}
+
+  handleError(error: Error) {
+    this.zone.run(
+      async () =>
+        await this.notification.swal
+          .fire({
+            icon: 'error',
+            text: error.message || 'Undefined client error',
```

Comment:
For errors I would prefer if we don't display the error message for the demo. Just say "Unexpected Error Occurred". Or something like that.